app/screens/gallery/renderers/video/hooks/index.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+import {runOnJS, runOnUI, useAnimatedReaction, type SharedValue} from 'react-native-reanimated';
+
+export function useStateFromSharedValue<T>(sharedValue: SharedValue<T> | undefined, defaultValue: T): T {
+    const [state, setState] = useState(defaultValue);
+    useAnimatedReaction(
+        () => sharedValue?.value,
+        (currentValue, previousValue) => {
+            if (currentValue !== previousValue) {
+                runOnJS(setState)(currentValue ?? defaultValue);
+            }
+        }, [defaultValue],
+    );
+
+    useEffect(() => {
+        if (sharedValue) {
+            runOnUI(() => {
+                'worklet';
+                const currentValue = sharedValue.value;
+                runOnJS(setState)(currentValue);
+            })();
+        }
+    }, [sharedValue]);
```

Comment:
Do these do the same thing?

Reply:
Great question! here, the `useEffect` and `useAnimatedReaction` serve different purposes, and is design to handle shared value synchronization.

Let me explain a bit more:
* `useAnimatedReaction` is used to handle ongoing changes to the share value, it triggers when the shared value have changed after the component is **mounted**, meaning that is reactive and only responds to future changes.
* `useEffect: this one will handle initial synchronization when the shared value **reference** changes, it triggers when *shareValue** prop/reference change (including on mount), in this case also triggers immediately as it mounts to capture the current state.

As long as the value of the *sharedValue* changes and not its reference, the useEffect should not trigger more than once.